eureka2-core/src/main/java/com/netflix/eureka2/channel/AbstractServiceChannel.java

```diff
@@ -60,4 +66,27 @@ public void onNext(T message) {
             }
         });
     }
+
+    protected boolean moveToState(STATE from, STATE to) {
+        if (state.compareAndSet(from, to)) {
+            if (metrics != null) {
+                // We do not track initState (==idle), only subsequent states that
+                // happen when a connection is established.
+                if (from == initState) {
+                    metrics.incrementStateCounter(to);
+                } else {
+                    metrics.stateTransition(from, to);
+                }
+            }
+            return true;
+        }
+        return false;
+    }
+
+    protected boolean moveToState(STATE to) {
+        if (state.get() == to) {
+            return false;
```

Comment:
What is the contract for the return of this method? If we are already at the state that we are trying to move to, it could be said that the moveToState is successful (i.e. true).